#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_xsp2_inherits.dpatch by Dylan R. E. Moonfire <debian@mfgames.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Temporarily corrects the problem with XSP2 using inherits working 
## DP: properly. This patch reverts that specific XSP2 code to XSP until 
## DB: it functions properly from the SVN.

@DPATCH@

diff -ru mono-1.1.9.1.orig/mcs/class/System.Web/System.Web.UI/TemplateParser.cs mono-1.1.9.1/mcs/class/System.Web/System.Web.UI/TemplateParser.cs
--- mono-1.1.9.1.orig/mcs/class/System.Web/System.Web.UI/TemplateParser.cs	2005-10-14 08:23:32.000000000 -0500
+++ mono-1.1.9.1/mcs/class/System.Web/System.Web.UI/TemplateParser.cs	2005-10-14 08:24:53.000000000 -0500
@@ -428,16 +428,13 @@
 				srcAssembly = GetAssemblyFromSource (src);
 
 			string inherits = GetString (atts, "Inherits", null);
-#if NET_2_0
-			className = inherits;
-#else
+
 			if (inherits != null)
 				SetBaseType (inherits);
 
 			className = GetString (atts, "ClassName", null);
 			if (className != null && !CodeGenerator.IsValidLanguageIndependentIdentifier (className))
 				ThrowParseException (String.Format ("'{0}' is not valid for 'className'", className));
-#endif
 
 			if (atts.Count > 0)
 				ThrowParseException ("Unknown attribute: " + GetOneKey (atts));
